src/Commands/core/UpdateDBCommands.php

```diff
@@ -187,15 +188,17 @@ public function process($batch_id, $options = ['format' => 'json'])
      *   The update number to run.
      * @param array $dependency_map
      *   The update dependency map.
+     * @param array $options
+     *   The update options.
      * @param DrushBatchContext $context
      *   The batch context object.
      */
-    public static function updateDoOne($module, $number, array $dependency_map, DrushBatchContext $context)
+    public static function updateDoOne($module, $number, array $dependency_map, array $options, DrushBatchContext $context)
     {
         $function = $module . '_update_' . $number;
 
         // Disable config entity overrides.
-        if (!defined('MAINTENANCE_MODE')) {
+        if ($options['maintenance-mode'] && !defined('MAINTENANCE_MODE')) {
             define('MAINTENANCE_MODE', 'update');
```

Comment:
Defining maintenance mode here does not have any effect on the Drupal site, as experienced through the web server. This definition only serves to make Drupal respond as if it were in maintenance mode during the update operation. By appearances, it looks like this code is designed specifically to operate when the site is NOT in maintenance model.

Perhaps this block, and the one like it below, should remain unmodified?  I am unsure about this. Do you have a specific reason or need to remove this?

Reply:
My concern is that once `define('MAINTENANCE_MODE', 'update');` is executed, all modules can assume that the site is in maintenance mode even in their *.install files which are loaded immediately after this code.

Reply:
Yes, all of the modules will think that the site is in maintenance mode, including when their update hooks run. This was added in #3606, and seems to still be necessary. I can't think of a bad effect of defining MAINTENANCE_MODE like this, but #3604 identifies situations where not defining it is problematic.